src/System.Private.CoreLib/shared/System/Numerics/Vector.cs

```diff
@@ -327,389 +324,25 @@ public unsafe Vector(T[] values, int index)
-                        *(basePtr + 5) = (short)(object)values[5 + index];
-                        *(basePtr + 6) = (short)(object)values[6 + index];
-                        *(basePtr + 7) = (short)(object)values[7 + index];
-                    }
-                }
-                else if (typeof(T) == typeof(uint))
-                {
-                    fixed (uint* basePtr = &this.register.uint32_0)
-                    {
-                        *(basePtr + 0) = (uint)(object)values[0 + index];
-                        *(basePtr + 1) = (uint)(object)values[1 + index];
-                        *(basePtr + 2) = (uint)(object)values[2 + index];
-                        *(basePtr + 3) = (uint)(object)values[3 + index];
-                    }
-                }
-                else if (typeof(T) == typeof(int))
-                {
-                    fixed (int* basePtr = &this.register.int32_0)
-                    {
-                        *(basePtr + 0) = (int)(object)values[0 + index];
-                        *(basePtr + 1) = (int)(object)values[1 + index];
-                        *(basePtr + 2) = (int)(object)values[2 + index];
-                        *(basePtr + 3) = (int)(object)values[3 + index];
-                    }
-                }
-                else if (typeof(T) == typeof(ulong))
-                {
-                    fixed (ulong* basePtr = &this.register.uint64_0)
-                    {
-                        *(basePtr + 0) = (ulong)(object)values[0 + index];
-                        *(basePtr + 1) = (ulong)(object)values[1 + index];
-                    }
-                }
-                else if (typeof(T) == typeof(long))
-                {
-                    fixed (long* basePtr = &this.register.int64_0)
-                    {
-                        *(basePtr + 0) = (long)(object)values[0 + index];
-                        *(basePtr + 1) = (long)(object)values[1 + index];
-                    }
-                }
-                else if (typeof(T) == typeof(float))
-                {
-                    fixed (float* basePtr = &this.register.single_0)
-                    {
-                        *(basePtr + 0) = (float)(object)values[0 + index];
-                        *(basePtr + 1) = (float)(object)values[1 + index];
-                        *(basePtr + 2) = (float)(object)values[2 + index];
-                        *(basePtr + 3) = (float)(object)values[3 + index];
-                    }
-                }
-                else if (typeof(T) == typeof(double))
-                {
-                    fixed (double* basePtr = &this.register.double_0)
-                    {
-                        *(basePtr + 0) = (double)(object)values[0 + index];
-                        *(basePtr + 1) = (double)(object)values[1 + index];
-                    }
-                }
-            }
+            this = Unsafe.ReadUnaligned<Vector<T>>(ref Unsafe.As<T, byte>(ref values[index]));
```

Comment:
If we're already using unsafe code, should we elide the `values[index]` as well?

Reply:
I will leave this as is for now. The bounds check was there before my change, they do not matter much since this is expanded as intrinsics (some of the time at least), and the whole thing becomes incomprehensible with the bounds check removed. This is candidate to fix up once we have proper public helpers to do this.